packages/admin/admin/src/form/FinalFormCurrencyInput.tsx

```diff
@@ -0,0 +1,62 @@
+import { InputBase } from "@mui/material";
+import * as React from "react";
+import { IntlShape, useIntl } from "react-intl";
+
+import { ClearInputAdornment } from "../common/ClearInputAdornment";
+import { FinalFormInputProps } from "./FinalFormInput";
+
+export function FinalFormCurrencyInput({
+    meta,
+    input,
+    innerRef,
+    clearable,
+    endAdornment,
+    disableContentTranslation,
+    currencySign,
+    currencySignPosition,
+    ...props
+}: FinalFormInputProps): React.ReactElement {
+    const intl = useIntl();
+    const [formattedCurrencyValue, setFormattedCurrencyValue] = React.useState("");
```

Comment:
Form Fields must use as prop passed `input` as SSOT and create it's own local state containing something created from value.

SSOT is a very important pattern you should always try to follow